### ISSUE TITLE
Mocha 6.2.2 binary does not exist when installed with yarn

### ISSUE THREAD

**Issue by ksmithut** [mochajs/mocha#4098]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

When installing mocha@6.2.2, the "mocha" command does not exist inside of `node_modules/.bin`

### Steps to Reproduce

```sh
> yarn --version
1.19.1
> node --version
v12.13.0
```

```sh
mkdir test
cd test
yarn init -y
yarn add -D mocha
ls -la node_modules/.bin/mocha # ls: cannot access 'node_modules/.bin/mocha': No such file or directory
```

If done with mocha@6.2.1, it exists as expected

**Expected behavior:** [What you expect to happen]

That the mocha binary would exist

**Actual behavior:** [What actually happens]

```
 ls: cannot access 'node_modules/.bin/mocha': No such file or directory
```

**Reproduces how often:** [What percentage of the time does it reproduce?]

Every time

### Versions

only fails for version 6.2.2 (latest at time of writing)

**Comment by ksmithut** [comment]
It wasn't clear in my original description, but using npm, it works as expected on every version.

**Comment by juergba** [comment]
There are a few issues in Yarn about this, I don't think this has anything to do with Mocha.
Please try: `yarn --check-files` or install Mocha a second time (without uninstall).

**Comment by ksmithut** [comment]
A subsequent install, nor the `yarn --check-files` worked. The thing that's add is that is works every time with mocha@6.2.1, but never with mocha@6.2.2

Additional info, upon installing, it appears the `node_modules/mocha/bin` directory have some differences between the two versions:

6.2.1:

```sh
> ls -la node_modules/mocha/bin  
total 24
drwxr-xr-x 2 [redacted] [redacted] 4096 Nov 20 16:05 .
drwxr-xr-x 6 [redacted] [redacted] 4096 Nov 20 16:05 ..
-rwxr-xr-x 1 [redacted] [redacted] 4565 Oct  8 08:46 mocha
-rwxr-xr-x 1 [redacted] [redacted]  185 Oct  8 08:46 _mocha
-rw-r--r-- 1 [redacted] [redacted]  306 Oct  8 08:46 options.js
```

6.2.2

```sh
ls -la node_modules/mocha/bin
total 12
drwxr-xr-x 2 [redacted] [redacted] 4096 Nov 20 16:06 .
drwxr-xr-x 5 [redacted] [redacted] 4096 Nov 20 16:06 ..
-rw-r--r-- 1 [redacted] [redacted]  306 Nov  1 16:18 options.js
```

Perhaps a dependency updated from 6.2.1 to 6.2.2 that skips some preinstall/postinstall script that creates those binaries? I've tried a few of the other solutions on the yarn issues to no avail. Below are the ones I found, but feel free to post others. Also feel free to close this issue because it seems to be a yarn issue.

**Comment by juergba** [comment]
@ksmithut thank you.

I tested on my Windows10 and the "node_modules/.bin" binaries are also missing. Additionally existing files in "node_modules/.bin" get deleted, so this folder remains empty at the end. There is an error message "EPERM: operation not permitted, unlink D:\.....\.bin\yarn".
The same result for mocha@6.2.1 and mocha@6.2.2.
I played around with the read-only attribute of "node_modules" folder, but without success.

I will keep open this issue for two weeks, then it will be closed. Having a look at yarn's issue queue tells me to do this.
